Task6/posts.js

```diff
@@ -0,0 +1,368 @@
+        hashTags: ['#love', '#lyrics'],
+        likes: ['Lola Skin', 'Paula', 'Egor B', 'Marie Curie', 'Dean W', 'Sam W']
+    },
+    {
+        id: '17',
+        description: 'The best known blues musician today is B.B. King. His fame is well-deserved.\n' +
+            '                Born in Indianola, Mississippi in 1925, he earned the nickname "B.B." ("Blues boy")\n' +
+            '                while playing on radio programs in Memphis, Tennessee.',
+        createdAt: new Date('2020-02-21T07:32:15'),
+        author: 'Alexander T',
+        photoLink: 'https://img.stereo.ru/article-covers/2018/cd94b715ef23dafa68adf62f2127b260.jpg',
+        hashTags: ['#king', '#enjoy', '#bluise'],
+        likes: ['Sean C', 'Egor B', 'Vlad K', 'Sam W']
+    },
+    {
+        id: '18',
+        description: 'There is nothing in the world so irresistibly contagious as laughter and good humor.',
+        createdAt: new Date('2020-02-21T10:14:00'),
+        author: 'Dickens',
+        hashTags: ['#reflections', '#thinkaboutit'],
+        likes: ['Lola', 'Paula', 'Egor B', 'Dean W', 'Sam W']
+    },
+    {
+        id: '19',
+        description: 'Kids, let\'s all get along!',
+        createdAt: new Date('2020-02-22T15:34:10'),
+        author: 'Leopold',
+        hashTags: ['#friendly', '#cat', '#sweety'],
+        likes: ['Kinnoske', 'Paula', 'Egor B', 'Dean W', 'Sam W']
+    },
+    {
+        id: '20',
+        description: 'America’s music culture would be incomplete without\n' +
+            '                blues music. Thought it was created in the early decades of the 20th century,\n' +
+            '                blues music has had a huge influence on American popular music\n' +
+            '                up to the present days. ',
+        createdAt: new Date('2020-02-22T20:40:00'),
+        author: 'Alexander T',
+        hashTags: ['#music', '#america', '#bluise'],
+        likes: ['Lola Skin', 'Paula', 'Egor B', 'Dean W', 'Sam W']
+    }
+];
+maxId = 20;
+
+function isInteger(num) {
+    return (num ^ 0) === num;
+}
+
+function sortByDateDescending(posts) {
+    posts.sort((p1, p2) => p1.createdAt < p2.createdAt ? 1 : -1);
+}
+
+function sortByDateAscending(posts) {
+    posts.sort((p1, p2) => p1.createdAt > p2.createdAt ? 1 : -1);
+}
+
+(function () {
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        if (!isInteger(parseInt(skip, 10)) || !(skip >= 0) || !isInteger(parseInt(top, 10)) || !(top > 0)) {
+            return 'Incorrect data';
+        }
+        let copy = posts.slice(0, posts.length), searching = [];
+        sortByDateDescending(copy);
+        if (filterConfig === undefined) {
+            for (let i = skip; i < skip + top; i++) {
+                searching.push(copy[i]);
+            }
+        } else {
+            if (filterConfig.hasOwnProperty('createdAt')) {
+                for (let i = skip; i < skip + top; i++) {
+                    if (copy[i].createdAt === filterConfig.createdAt) {
```

Comment:
It should filter not by equality, but somehow like "created after date1, before date2"